%YAML:1.0

# for EuRoc cam0

#Rotation from camera frame to imu frame, imu^R_cam
extrinsicRotation: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [0.01486, -0.99988, 0.00414, 
           0.999557, 0.014976, 0.0257155, 
           -0.02577, 0.00375, 0.99966]
#Translation from camera frame to imu frame, imu^T_cam
extrinsicTranslation: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [-0.0216401,-0.0646770, 0.0098107]

# Camera calibration and distortion parameters (OpenCV)
Camera.model: "pinhole" # "pinhole" or "atan"
Camera.fx: 458.654
Camera.fy: 457.296
Camera.cx: 367.215
Camera.cy: 248.375

Camera.k1: -0.28340811    # parameters for "pinhole" camera
Camera.k2: 0.07395907
Camera.p1: 0.00019359
Camera.p2: 1.76187114e-05

# Camera frames per second
Camera.fps: 20.0

#imu
acc_n: 0.2          # accelerometer measurement noise standard deviation. #0.2
gyr_n: 0.02         # gyroscope measurement noise standard deviation.     #0.05
acc_w: 0.0002         # accelerometer bias random work noise standard deviation.  #0.02
gyr_w: 2.0e-5       # gyroscope bias random work noise standard deviation.     #4.0e-5

# Image
IMAGE_TOPIC: "/cam0/image_raw"
IMU_TOPIC: "/imu0"
Image.width: 752
Image.height: 480
Image.pyramid_levels: 3  # 0-based

# Fast detector
FastDetector.grid_size: 64
FastDetector.grid_min_size: 8
FastDetector.fast_max_threshold: 20
FastDetector.fast_min_threshold: 7
FastDetector.fast_min_eigen: 30.0

# Initialization
Initializer.min_corners: 200
Initializer.min_tracked: 150
Initializer.min_inliers: 80
Initializer.min_disparity: 30
Initializer.sigma: 1.0
Initializer.ransac_max_iters: 1000

# Map
Mapping.scale: 1
Mapping.min_connection_observations: 5 # min shared fts in connected kfs
Mapping.min_corners: 200 # min corners per keyframe
Mapping.max_reproject_kfs: 25
Mapping.max_local_ba_kfs: 10
Mapping.min_local_ba_connected_fts: 20

# Align
Align.top_level: 3   # not bigger than Image.pyramid_levels
Align.bottom_level: 0   # not smaller than 0
Align.patch_size: 4

# Tracking
Tracking.max_local_kfs: 10
Tracking.min_quality_fts: 5
Tracking.max_quality_drop_fts: 40

# DepthFilter
DepthFilter.max_perprocess_kfs: 3
DepthFilter.max_seeds_buffer: 20

# glog
Glog.alsologtostderr: 1
Glog.colorlogtostderr: 1
Glog.stderrthreshold: 1  # Copy log messages at or above this level to stderr in addition to logfiles. The numbers of severity levels INFO, WARNING, ERROR, and FATAL are 0, 1, 2, and 3, respectively.
Glog.minloglevel: 1      # Log messages at or above this level. Again, the numbers of severity levels INFO, WARNING, ERROR, and FATAL are 0, 1, 2, and 3, respectively.
Glog.log_prefix: 0       # Set whether the log prefix should be prepended to each line of output.
Glog.log_dir: "" # If specified, logfiles are written into this directory instead of the default logging directory.

# Trace log
Trace.log_dir: "/tmp"

# DBoW
DBoW.voc_dir: ""
